csv/write.go

```diff
@@ -7,21 +7,41 @@ import (
 
 	"github.com/apache/arrow/go/v13/arrow"
 	"github.com/apache/arrow/go/v13/arrow/csv"
+	"github.com/cloudquery/filetypes/v2/types"
 )
 
-func (cl *Client) WriteTableBatch(w io.Writer, arrowSchema *arrow.Schema, records []arrow.Record) error {
-	writer := csv.NewWriter(w, arrowSchema,
+type Handle struct {
+	w *csv.Writer
+	s *arrow.Schema
+}
+
+var _ types.Handle = (*Handle)(nil)
+
+func (cl *Client) WriteHeader(w io.Writer, s *arrow.Schema) (types.Handle, error) {
```

Comment:
In this particular format/package, the csv header is written with the first call to `.Write` (and not in this fn) so the name might be misleading. Maybe rename the interface method to 'Begin' etc?